imap/jmap_ical.c

```diff
@@ -1627,6 +1628,57 @@ static int identity_int(int i) {
 
 static json_t* relatedto_from_ical(icalcomponent*);
 
+static int is_reserved_jicalprop(icalcomponent_kind kind, const char *name)
+{
+    // all IANA property names are reserved
+    if (strncasecmpsafe(name, "X-", 2))
+        return 1;
+
+    // all JMAP extension property names are reserved
+    if (!strncasecmpsafe(name, "X-JMAP", 6))
+        return 1;
+
+    // we used this for useDefaultAlerts on VEVENTs
+    if (!strcasecmpsafe(name, "X-APPLE-DEFAULT-ALARM") &&
+            kind == ICAL_VEVENT_COMPONENT)
```

Comment:
Probably a little nit-picky, but I would do the component type check first since its cheaper than a string comparison.  No need to do the string check if we're going to ignore it.

Reply:
I know and actually had written this first the way you suggested. I figured the way it's currently written is more consistent to read with the other conditionals, and the cost of the string comparison is negligible. But since you noticed, I'll rewrite it the previous variant.